### PR TITLE
Add timeouts to requests in Freshmaker

### PR DESCRIPTION
**Pull request by ElenaKarolinaSemanova**
Add configurable timeout to requests to avoid FM from hanging.

### CODE REVIEW

**Comment by ElenaKarolinaSemanova** [comment]
@qixiang , @apaplaus, please review.

File: freshmaker/lightblue.py [hunk 1]
```diff
@@ -581,9 +581,9 @@ def _make_request(self, entity, data):
                 record_mode=conf.vcrpy_mode,
             )
             with my_vcr.use_cassette(f'{self.event_id}.yml'):
-                response = requests.post(entity_url, **request_kwargs)
+                response = requests.post(entity_url, **request_kwargs, timeout=conf.requests_timeout)
         else:
-            response = requests.post(entity_url, **request_kwargs)
+            response = requests.post(entity_url, **request_kwargs, timeout=conf.requests_timeout)
```
Comment (qixiang) [inline freshmaker/lightblue.py:586]: Requests to LightBlue can take much more time, so we don't want to apply the same timeout to LightBlue requests as others, I believe some LightBlue requests can even take more than 5mins to finish.

**Comment by apaplaus** [comment]
LGFM, just address the comment from qwan

File: contrib/freshmaker-cli [hunk 2]
```diff
@@ -181,7 +182,7 @@ def _get_freshmaker_items(url, params, limit=None):
     ret_items = []
     while True:
         print("Getting messages from Freshmaker, page", params["page"])
-        r = requests.get(url, verify=False, params=params)
+        r = requests.get(url, verify=False, params=paramsi, timeout=conf.requests_timeout)
```
Comment (mprahl) [inline contrib/freshmaker-cli:185]:
```suggestion
        r = requests.get(url, verify=False, params=params, timeout=conf.requests_timeout)
```

File: freshmaker/lightblue.py (hunk 1 above)
Comment (ElenaKarolinaSemanova) [reply freshmaker/lightblue.py]: @qixiang I fixed the timeout for LightBlue. Review one more time, please.

File: contrib/freshmaker-cli (hunk 2 above)
Comment (ElenaKarolinaSemanova) [reply contrib/freshmaker-cli]: Fixed.

File: freshmaker/lightblue.py [hunk 3]
```diff
@@ -581,9 +581,9 @@ def _make_request(self, entity, data):
                 record_mode=conf.vcrpy_mode,
             )
             with my_vcr.use_cassette(f'{self.event_id}.yml'):
-                response = requests.post(entity_url, **request_kwargs)
+                response = requests.post(entity_url, **request_kwargs, timeout=conf.requests_timeout)
         else:
-            response = requests.post(entity_url, **request_kwargs)
+            response = requests.post(entity_url, **request_kwargs, timeout=300)
```
Comment (qixiang) [inline freshmaker/lightblue.py:586]: Not sure if 300s is good enough, we can have another timeout variable in conf for lightblue requests, so we can easily adjust it if we see any issues, or maybe:
```timeout=max(600, conf.requests_timeout*5)```

Comment (ElenaKarolinaSemanova) [reply freshmaker/lightblue.py]: @qixiang Fixed it, you can take a look

**Comment by qixiang** [comment]
LGTM

**Pull request merged**